docs/tutorials/basic/data.md

```diff
@@ -187,22 +211,361 @@ class SimpleIter:
+plt.imshow(tmp.asnumpy()); plt.show()
+```
+
+    (80, 128, 150, 200)
+
+![](https://github.com/dmlc/web-data/blob/master/mxnet/doc/tutorials/basic/data/output_45_1.png)
+
+### Loading Data using Image Iterators
+
+Before we see how to read data using the two built-in Image Iterators,
+ lets get a sample dataset __Caltech 101__ dataset
+ that contains 101 classes of objects and converts them into record io format.
+Download and unzip
+
+
+```python
+os.system('wget http://www.vision.caltech.edu/Image_Datasets/Caltech101/101_ObjectCategories.tar.gz -P data/')
+os.chdir('data')
+os.system('tar -xf 101_ObjectCategories.tar.gz')
+os.chdir('../')
+```
+
+Let's take a look at the data. As you can see, under the [root folder](./data/101_ObjectCategories) every category has a [subfolder](./data/101_ObjectCategories/yin_yang).
+
+Now let's convert them into record io format using the `im2rec.py` utility scipt.
+First we need to make a list that contains all the image files and their categories:
+
+
+```python
+os.system('python %s/tools/im2rec.py --list=1 --recursive=1 --shuffle=1 --test-ratio=0.2 data/caltech data/101_ObjectCategories'%MXNET_HOME)
+```
+
 
-- `image_iter` by `ImageRecordIter` so we can enjoy the provided multithreaded
-pre-fetch and augmentation.
-- `caption_iter` by `NDArrayIter` or bucketing iterator provided in the `rnn`
-package.
 
-Next we create an iterator whose `next()` function will both
-`image_iter.next()` and `caption_iter.next()` and return the combined results.
+The resulting [list file](./data/caltech_train.lst) is in the format `index\t(one or more label)\tpath`. In this case there is only one label for each image but you can modify the list to add in more for multi label training.
 
+Then we can use this list to create our record io file:
+
+
+```python
+os.system("python %s/tools/im2rec.py --num-thread=4 --pass-through=1 data/caltech data/101_ObjectCategories"%MXNET_HOME)
+```
+
+The record io files are now saved at [here](./data)
+
+#### Using ImageIter
+[__ImageIter__](http://mxnet.io/api/python/io.html#mxnet.io.ImageIter) is a flexible interface that supports loading of images from both in RecordIO and Raw format.
+
+
+```python
+data_iter = mx.image.ImageIter(batch_size=4, data_shape=(3, 227, 227),
+                              path_imgrec="./data/caltech.rec",
+                              path_imgidx="./data/caltech.idx" )
+data_iter.reset()
+batch = data_iter.next()
+data = batch.data[0]
+for i in range(4):
+    plt.subplot(1,4,i+1)
+    plt.imshow(data[i].asnumpy().astype(np.uint8).transpose((1,2,0)))
+plt.show()
+```
+
+    loading recordio...
+
+![](https://github.com/dmlc/web-data/blob/master/mxnet/doc/tutorials/basic/data/output_55_1.png)
+
+
+#### Using ImageRecordIter
+[__`ImageRecordIter`__](http://mxnet.io/api/python/io.html#mxnet.io.ImageRecordIter) can be used for loading image data saved in record io format. To use ImageRecordIter, simply create an instance by loading your record file:
```

Comment:
put ImageRecordIter first. Note its faster but restrictive. Use ImageIter if you want flexibility.